src/lib/mech.c

```diff
@@ -708,6 +709,92 @@ CK_RV rsa_pkcs_synthesizer(CK_MECHANISM_PTR mech, attr_list *attrs,
     return CKR_OK;
 }
 
+CK_RV rsa_pss_synthesizer(CK_MECHANISM_PTR mech, attr_list *attrs,
+        CK_BYTE_PTR inbuf, CK_ULONG inlen,
+        CK_BYTE_PTR outbuf, CK_ULONG_PTR outlen) {
+
+    const EVP_MD *md = NULL;
+    CK_RV rv = mech_get_digester(mech, &md);
+    if (rv != CKR_OK) {
+        LOGE("Could not get digester for mech: 0x%lx", mech->mechanism);
+        return rv;
+    }
+
+    int expected_len = EVP_MD_size(md);
+    if (expected_len <= 0) {
+        LOGE("Hash size cannot be 0 or negative, got: %d",
+                expected_len);
+        return CKR_GENERAL_ERROR;
+    }
+
+    if (inlen != (unsigned)expected_len) {
+        LOGE("Expected input size to be hash size, %lu != %d",
+                inlen, expected_len);
+        return CKR_GENERAL_ERROR;
+    }
+
+    CK_ATTRIBUTE_PTR modulus_attr = attr_get_attribute_by_type(attrs, CKA_MODULUS);
+    if (!modulus_attr) {
+        LOGE("Signing key has no CKA_MODULUS");
+        return CKR_GENERAL_ERROR;
+    }
+
+    CK_ATTRIBUTE_PTR exp_attr = attr_get_attribute_by_type(attrs, CKA_PUBLIC_EXPONENT);
+    if (!exp_attr) {
+        LOGE("Signing key has no CKA_PUBLIC_EXPONENT");
+        return CKR_GENERAL_ERROR;
+    }
+
+    if (modulus_attr->ulValueLen > *outlen) {
+        LOGE("Output buffer is too small, got: %lu, required at least %lu",
+                *outlen, modulus_attr->ulValueLen);
+        return CKR_GENERAL_ERROR;
+    }
+
+    RSA *rsa = RSA_new();
+    if (!rsa) {
+        LOGE("oom");
```

Comment:
Can lean rsa

Reply:
We can also drop the exponent, as we don't need it. Internally the code just needs n.

Reply:
Nope can't drop e, set0key will fail.